module/zfs/arc.c

```diff
@@ -10442,7 +10445,9 @@ l2arc_log_blk_commit(l2arc_dev_t *dev, zio_t *pio, l2arc_write_callback_t *cb)
 
 	/* try to compress the buffer */
 	psize = zio_compress_data(ZIO_COMPRESS_LZ4,
-	    abd_buf->abd, (void **) &tmpbuf, sizeof (*lb), 0);
+	    abd_buf->abd, (void **) &tmpbuf, sizeof (*lb),
+	    sizeof (*lb) - sizeof (*lb) % (1 << dev->l2ad_vdev->vdev_ashift),
```

Comment:
With `sizeof (*lb)` always being 64KB the result will always be `sizeof (*lb)`.  Not enough.